### PR TITLE
JavaScript: Add concrete syntax examples to `JSDoc`, `RegExp`, `Tokens`, `Variables` and `XML`.

### PR DESCRIPTION
**Pull request by xiemaisi**
I think this covers most of the remaining AST classes, except for TypeScript.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/Regexp.qll [hunk 1]
```diff
@@ -160,68 +223,193 @@ class RegExpSequence extends RegExpTerm, @regexp_seq {
...
 }
 
-/** A range-quantified term, that is, a term of the form `...{m,n}`. */
+/**
+ * A range-quantified term
+ *
+ * Example:
+ *
+ * ```
+ * [0-9a-fA-F]{2,4}
```
Comment (asger-semmle) [inline javascript/ql/src/semmle/javascript/Regexp.qll]: Unfortunate example, given that `a-f` and `A-F` are also "ranges". Maybe just pick a simpler base expression.

File: javascript/ql/src/semmle/javascript/Variables.qll [hunk 2]
```diff
@@ -669,6 +823,15 @@ class DeclarationSpace extends string {
  * A name that is declared in a particular scope.
  *
  * This can be a variable or a local name for a TypeScript type or namespace.
+ *
+ * Examples:
+ *
+ * ```
+ * // `any`, `id` and `x` are lexical names
+ * function id(x: number) : any {
```
Comment (asger-semmle) [inline javascript/ql/src/semmle/javascript/Variables.qll]: Wait, `any` is a keyword, not a name.

File: javascript/ql/src/semmle/javascript/Variables.qll [hunk 3]
```diff
@@ -690,6 +853,14 @@ class LexicalName extends @lexical_name {
 
 /**
  * An identifier that refers to a variable, type, or namespace, or a combination of these.
+ *
+ * Example:
+ *
+ * ```
+ * function id(x: number) : any {  // `id`, `x`, `number` and `any` are lexical references
```
Comment (asger-semmle) [inline javascript/ql/src/semmle/javascript/Variables.qll]: ditto

File: javascript/ql/src/semmle/javascript/Variables.qll [hunk 4]
```diff
@@ -710,11 +881,27 @@ class LexicalRef extends Identifier, @lexical_ref {
...
 class LexicalDecl extends LexicalRef, @lexical_decl { }
 
 /**
  * An identifier that refers to a variable, type, or namespace, or a combination of these,
  * in a non-declaring position.
+ *
+ * Example:
+ *
+ * ```
+ * function id(x: number) : any {  // `number` and `any` are lexical  accesses
```
Comment (asger-semmle) [inline javascript/ql/src/semmle/javascript/Variables.qll]: ditto

**Comment by xiemaisi** [comment]
Comments addressed.

**Pull request merged**